server/src/main/java/io/crate/execution/ddl/tables/AddColumnRequest.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import io.crate.analyze.AnalyzedColumnDefinition;
+import io.crate.metadata.GeoReference;
+import io.crate.metadata.IndexType;
+import io.crate.metadata.Reference;
+import io.crate.metadata.RelationName;
+import io.crate.sql.tree.CheckColumnConstraint;
+import io.crate.sql.tree.ColumnPolicy;
+import io.crate.types.ArrayType;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AddColumnRequest extends AcknowledgedRequest<AddColumnRequest> {
+
+    private final RelationName relationName;
+
+    // Used only with ADD COLUMN.
+    // Not included into StreamableColumnInfo to avoid adding empty list in most cases.
+    // Checks are merged into mapping without knowing column name as it's contained in expression itself
+    // and thus we can keep them separately which is not the case for other constraints.
+    private final List<StreamableCheckConstraint> checkConstraints = new ArrayList<>();
```

Comment:
Does this consider that there are both column level checks and table level checks?

Reply:
afaik, on ALTER TABLE it's always column level check, there is even a validation in analyzer
```
// SQL Semantics: CHECK expressions cannot refer to other
// columns to not invalidate existing data inadvertently.
```
but in terms of mapping its anyway same format, I debugged create table statement 
with column level check
```
  execute("create table t (" +
                      "    id int primary key, " +
                      "    qty int constraint check_qty_gt_zero check(qty > 0), " +
                      "    constraint check_id_ge_zero check (id >= 0)" +
                      ")");
```
part of the request: `"check_constraints":{"check_qty_gt_zero":"\"qty\" > 0","check_id_ge_zero":"\"id\" >= 0"}}`
also `check_name->expression`, so for this code it doesn't make any difference